Classes/Utility/FrontendUtility.php

```diff
@@ -185,21 +185,21 @@ public static function isCacheable($conf = null)
     /**
      * Return current URL
      *
-     * @return null|string
+     * @return string
      */
     public static function getCurrentUrl()
     {
-        $ret = null;
-
-        $TSFE = self::getTsfe();
+        //former $TSFE->anchorPrefix got deprecated in TYPO3 7.x
+        $tsfeAnchorPrefix = substr(
+            Typo3GeneralUtility::getIndpEnv('TYPO3_REQUEST_URL'),
```

Comment:
Revisiting this, I come to the conclusion that `'TYPO3_REQUEST_URL'` not equals `'TYPO3_SITE_URL'`.

Would be wrong to use $requestUrl twice here.